scripts/tools/InitializeQueue.php

```diff
@@ -188,8 +160,50 @@ private function checkParams(array $params)
             }
         }
 
-        if ($this->wantedBroker == self::BROKER_RDS && !$this->persistenceId) {
+        $this->validateBrokersWithPersistence();
+    }
+
+    private function validateBrokersWithPersistence(): void
```

Comment:
- [ ] This validation can be inside the factory, since is part of building the broker.

Reply:
i agree ,  the only part is that were don't have integration tests , thus making this changes on a brittle extension should be limited , i will move this tot he brokerFactory

Reply:
I asked cause it is new code.